### ISSUE TITLE
Package: conditions/convert-comparison-to-boolean - object properties marked as constant

### ISSUE THREAD

**Issue by epreston** [coderaiser/putout#140]
Package: conditions/convert-comparison-to-boolean
Issue: non constant members are seen as constant by this rule

Example: 

```js
// constant object reference
const constObjRef = {
  notConstantProp: 0,
}

// we can update the property at any point
constObjRef.notConstantProp = 100;

// this rule sees this as a constant comparison: 0 > 40
// it is actually a comparison of: 100 > 40
if (constObjRef.notConstantProp > 40) {
  // do something
}

// the rule wants to replace it with
if (false) {
  // do something
}

// but the actual result is true during execution
if (true) {
  // do something
}

// it should just leave this alone because it can be updated at any time
if (constObjRef.notConstantProp > 40) {
  // do something
}
```

I am guessing that this rule sees the constant declaration of the reference to the object and treats all members as constant.

**Comment by epreston** [comment]
Retesting for 29.4.0 update.

**Comment by coderaiser** [comment]
It is related to:

https://github.com/coderaiser/putout/blob/aec7696bdfadac022f760e006220bc80b3680105/packages/operate/lib/compute.js#L52L75

Do you have ideas how to fix?

**Comment by epreston** [comment]
The way I would fix is by excluding property accessors from this rule.

I've only given it 10 minutes thought but, I can't think of how these would can be made constant.

**Comment by epreston** [comment]
Even a typescript enum gets boiled down to a mutable object.

Typescript example:

```ts
enum Direction {
  Up,
  Down,
  Left,
  Right
}
```

Gets converted to this javascript

```js
const Direction = {
  Up: 'Up',
  Down: 'Down',
  Left: 'Left',
  Right: 'Right'
};
```

Only things know of is:
- abusing classes and static members https://2ality.com/2020/01/enum-pattern.html
- or using a less sophisticated class with object.freeze

**Comment by epreston** [comment]
Issue might be with:

https://github.com/coderaiser/putout/blob/aec7696bdfadac022f760e006220bc80b3680105/packages/operate/lib/compute.js#L84

**Comment by coderaiser** [comment]
`compute` tries to determine value of AST node, you can change the code and check if it affects all 🐊**Putout** tests, if not we can change implementation.

But better to start from #139, it is much simpler.

**Comment by epreston** [comment]
Sounds good.  I'll start from there.

**Comment by coderaiser** [comment]
Fixed with 20f19c8, landed in `@putout/operate@8.10.0` 🎉 . Is it works for you?
